Non-root users cannot run backend tests using logical/testing.Test()

CoreConfig passed to NewCore method does not disable mlock.

testing.go:131: error initializing core:  Failed to lock memory: cannot allocate memory
		
		This usually means that the mlock syscall is not available.
		Vault uses mlock to prevent memory from being swapped to
		disk. This requires root privileges as well as a machine
		that supports mlock. Please enable mlock on your system or
		disable Vault from using it. To disable Vault from using it,
		set the `disable_mlock` configuration option in your configuration
		file.

--- a/logical/testing/testing.go
+++ b/logical/testing/testing.go
@@ -126,6 +126,7 @@ func Test(t TestT, c TestCase) {
 				return c.Factory(conf)
 			},
 		},
+		DisableMlock: true,
 	})
 	if err != nil {
 		t.Fatal("error initializing core: ", err)

